CMSgov/mdct-mcr
# reduce sanitizer, handle for safe types, add additional tests

## Description

@scwambach and I had attempted to reduce our API sanitization methods but ran into some tricky parts. This PR completes that work, adds handling for trusted safe types (number, boolean, undefined, null, NaN) and adds a few more tests 'cause who doesn't love more test coverage?

### How to test

1. Run the tests and celebrate their passing.
2. Confirm that calls to the database still pass nice data, and even strip out bad data.

### Changed Dependencies

n/a

## Code author checklist

## Reviewer checklist (two different people)
